eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Genesis.hs

```diff
@@ -100,3 +125,211 @@ instance ToCBOR AlonzoGenesis where
+    _ -> toJSON r
+
+instance ToJSON Prices where
+  toJSON Prices {prSteps, prMem} =
+    -- We cannot round-trip via NonNegativeInterval, so we go via Rational
+    object
+      [ "prSteps" .= toRationalJSON (BT.unboundRational prSteps),
+        "prMem" .= toRationalJSON (BT.unboundRational prMem)
+      ]
+
+instance FromJSON Prices where
+  parseJSON =
+    Aeson.withObject "prices" $ \o -> do
+      steps <- o .: "prSteps"
+      mem <- o .: "prMem"
+      prSteps <- checkBoundedRational steps
+      prMem <- checkBoundedRational mem
+      return Prices {prSteps, prMem}
+    where
+      -- We cannot round-trip via NonNegativeInterval, so we go via Rational
+      checkBoundedRational r =
+        case BT.boundRational r of
+          Nothing -> fail ("too much precision for bounded rational: " ++ show r)
+          Just s -> return s
+
+deriving newtype instance FromJSON CostModel
+
+deriving newtype instance ToJSON CostModel
+
+languageToText :: Language -> Text
+languageToText PlutusV1 = "PlutusV1"
+languageToText PlutusV2 = "PlutusV2"
+
+languageFromText :: MonadFail m => Text -> m Language
+languageFromText "PlutusV1" = pure PlutusV1
+languageFromText lang = fail $ "Error decoding Language: " ++ show lang
+
+instance FromJSON Language where
+  parseJSON = Aeson.withText "Language" languageFromText
+
+instance ToJSON Language where
+  toJSON = Aeson.String . languageToText
+
+instance ToJSONKey Language where
+  toJSONKey = toJSONKeyText languageToText
+
+instance FromJSONKey Language where
+  fromJSONKey = Aeson.FromJSONKeyTextParser languageFromText
+
+instance FromJSON AlonzoGenesis where
+  parseJSON = Aeson.withObject "Alonzo Genesis" $ \o -> do
+    coinsPerUTxOWord <-
+      o .: "lovelacePerUTxOWord"
+        <|> o .: "adaPerUTxOWord" --TODO: deprecate
+    cModels <- o .:? "costModels"
+    prices <- o .: "executionPrices"
+    maxTxExUnits <- o .: "maxTxExUnits"
+    maxBlockExUnits <- o .: "maxBlockExUnits"
+    maxValSize <- o .: "maxValueSize"
+    collateralPercentage <- o .: "collateralPercentage"
+    maxCollateralInputs <- o .: "maxCollateralInputs"
+    case cModels of
+      Nothing -> case CostModel <$> defaultCostModelParams of
+        Just m ->
+          return
+            AlonzoGenesis
+              { coinsPerUTxOWord,
+                costmdls = Map.singleton PlutusV1 m,
+                prices,
+                maxTxExUnits,
+                maxBlockExUnits,
+                maxValSize,
+                collateralPercentage,
+                maxCollateralInputs
+              }
+        Nothing -> fail "Failed to extract the cost model params from defaultCostModel"
+      Just costmdls ->
+        return
+          AlonzoGenesis
+            { coinsPerUTxOWord,
+              costmdls,
+              prices,
+              maxTxExUnits,
+              maxBlockExUnits,
+              maxValSize,
+              collateralPercentage,
+              maxCollateralInputs
+            }
+
```

Comment:
@Jimbo4350 - in node there is a comment here stating that the cost model is rendered separately. but below I seed the cost models being rendered. can I delete this comment?

https://github.com/input-output-hk/cardano-node/blob/42582c828ac4c364839427bc5bd604a03c9ac6b3/cardano-api/src/Cardano/Api/Orphans.hs#L416-L418

Reply:
Yes this comment is outdated and now incorrect, you can delete it :+1: .